libbpf-tools/llcstat.bpf.c

```diff
@@ -3,36 +3,42 @@
 #include <vmlinux.h>
 #include <bpf/bpf_helpers.h>
 #include <bpf/bpf_tracing.h>
+#include "maps.bpf.h"
 #include "llcstat.h"
 
 #define MAX_ENTRIES	10240
 
+const volatile bool targ_per_thread = false;
+
 struct {
 	__uint(type, BPF_MAP_TYPE_HASH);
 	__uint(max_entries, MAX_ENTRIES);
-	__type(key, u64);
-	__type(value, struct info);
+	__type(key, struct key_info);
+	__type(value, struct value_info);
 } infos SEC(".maps");
 
 static __always_inline
 int trace_event(__u64 sample_period, bool miss)
 {
-	u64 pid = bpf_get_current_pid_tgid();
-	u32 cpu = bpf_get_smp_processor_id();
-	struct info *infop, info = {};
-	u64 key = pid << 32 | cpu;
-
-	infop = bpf_map_lookup_elem(&infos, &key);
-	if (!infop) {
-		bpf_get_current_comm(info.comm, sizeof(info.comm));
-		infop = &info;
-	}
+	struct key_info key = {};
+	struct value_info *infop, zero = {};
+
+	key.cpu = bpf_get_smp_processor_id();
+	key.pid = bpf_get_current_pid_tgid() >> 32;
+	if (targ_per_thread)
+		key.tid = (u32)bpf_get_current_pid_tgid();
+	else
+		key.tid = key.pid;
```

Comment:
`key.tid = 0;` ?

Reply:
if not set targ_per_thread, do not use key.tid filed, default key.tid = key.pid